src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.VsLanguageDebugInfo.cs

```diff
@@ -278,7 +278,7 @@ public int ValidateBreakpointLocation(IVsTextBuffer pBuffer, int iLine, int iCol
                 if (textBuffer != null)
                 {
                     var snapshot = textBuffer.CurrentSnapshot;
-                    Document document = snapshot.AsText().GetDocumentWithFrozenPartialSemanticsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                    Document document = snapshot.AsText().GetDocumentWithFrozenPartialSemantics(cancellationToken);
```

Comment:
Blocking on async on a UI thread operation.  This is likely why people might have seen breakpoint resolution modal dialogs lasting a long time.